### PR TITLE
try to get_role, but fall back to role from regex

### PR DESCRIPTION
**Pull request by andremoeller**
*Issue #, if available:* https://github.com/aws/sagemaker-python-sdk/issues/291

*Description of changes:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/session.py [hunk 1]
```diff
@@ -763,7 +763,10 @@ def get_caller_identity_arn(self):
 
         # Call IAM to get the role's path
         role_name = role[role.rfind('/') + 1:]
-        role = self.boto_session.client('iam').get_role(RoleName=role_name)['Role']['Arn']
+        try:
+            role = self.boto_session.client('iam').get_role(RoleName=role_name)['Role']['Arn']
+        except ClientError:
+            LOGGER.warning("Couldn't call 'get_role' to get Role ARN from role name {}.".format(role_name))
```
Comment (laurenyu) [inline src/sagemaker/session.py]: I'd include a note here that this will exclude a role's path if it has one

File: tests/unit/test_session.py [hunk 2]
```diff
@@ -69,6 +69,16 @@ def test_get_caller_identity_arn_from_an_user(boto_session):
...
 
 
+def test_get_caller_identity_arn_from_an_user_without_permissions(boto_session):
+    sess = Session(boto_session)
+    arn = 'arn:aws:iam::369233609183:user/mia'
+    sess.boto_session.client('sts').get_caller_identity.return_value = {'Arn': arn}
+    sess.boto_session.client('iam').get_role.side_effect = ClientError('Bad permissions!', {})
+
+    actual = sess.get_caller_identity_arn()
+    assert actual == 'arn:aws:iam::369233609183:user/mia'
```
Comment (laurenyu) [inline tests/unit/test_session.py]: would it be possible to also test if the logger does warn?

File: src/sagemaker/session.py (hunk 1 above)
Comment (harusametime) [reply src/sagemaker/session.py]: @andremoeller @laurenyu 
*AmazonSageMakerFullAccess* that is a managed policy for SageMaker \(see [doc](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-amazonsagemakerfullaccess-policy)\) does not allow `iam:GetRole`.  The line in `try`always fail and goes to `except` if users attach only *AmazonSageMakerFullAccess* to the role; `get_role()` does not work. Could you consider another implementation for this?

Comment (andremoeller) [reply src/sagemaker/session.py]: @harusametime 

You're right -- in that case. The purpose of this is to attempt to get the role using `iam:GetRole`, but never to fail in this case -- not all users will only be using `AmazonSageMakerFullAccess` policy.

**Pull request merged**